File: app/src/main/java/mb00/android/codehub/api/model/Gist.java
package mb00.android.codehub.api.model;

import com.google.gson.annotations.SerializedName;

import java.util.Map;


public class Gist {

    @SerializedName("id")
    private String id;
    @SerializedName("files")
    private Map<String, GistFile> files;
    @SerializedName("description")
    private String description;
    @SerializedName("created_at")
    private String creationDate;
    @SerializedName("updated_at")
    private String updateDate;
    @SerializedName("comments")
    private int commentCount;
    @SerializedName("owner")
    private Owner owner;

    public String getId() {
        return id;
    }

    public Map<String, GistFile> getFiles() {
        return files;
    }

    public String getDescription() {
        return description;
    }

    public String getCreationDate() {
        return creationDate;
    }

    public String getUpdateDate() {
        return updateDate;
    }

    public int getCommentCount() {
        return commentCount;
    }

    public Owner getOwner() {
        return owner;
    }

}

File: app/src/main/java/mb00/android/codehub/api/model/Gist.kt
package mb00.android.codehub.api.model

import com.google.gson.annotations.SerializedName


data class Gist(
        @SerializedName("id") var id: String?,
        @SerializedName("files") var files: Map<String, GistFile>?,
        @SerializedName("description") var description: String?,
        @SerializedName("created_at") var creationDate: String?,
        @SerializedName("updated_at") var updateDate: String?,
        @SerializedName("comments") var commentCount: Int?,
        @SerializedName("owner") var owner: Owner?
)
